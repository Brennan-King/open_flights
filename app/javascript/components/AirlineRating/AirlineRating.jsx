import React from 'react';
import PropTypes from 'prop-types';
import './AirlineRating.scss';

const propTypes = {
  /** The score of the review */
  reviewScore: PropTypes.number,
};

const propDefaults = {
  reviewScore: '',
};

/**
 * Constructs an airline rating component used to display the
 * rating of a specific airline.
 * @param {object} props the props used to configure the airline rating
 * @return {node} a div containing the configured airline rating
 */
const AirlineRating = (props) => {
  const {reviewScore} = props;

  const buildAirlinelineRating = () => {
    const airlineRatings = [];

    for (let scoreIndex = 0; scoreIndex < reviewScore; scoreIndex++) {
      airlineRatings.push(
          <div className="airline-rating-filled-in" />,
      );
    }
    for (let scoreIndex = 0; scoreIndex < 5 - reviewScore; scoreIndex++) {
      airlineRatings.push(
          <div className="airline-rating-not-filled-in" />,
      );
    }

    return (airlineRatings);
  };

  return (
    <span className="airline-rating-wrapper">
      {buildAirlinelineRating()}
    </span>
  );
};

AirlineRating.propTypes = propTypes;
AirlineRating.propDefaults = propDefaults;
export default AirlineRating;
