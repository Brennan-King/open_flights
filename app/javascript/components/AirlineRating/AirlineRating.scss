.airline-rating-wrapper {
  display: flex;
}

.airline-rating-filled-in {
  height: 25px;
  width: 25px;
  background-color: #fcc201;
  border-radius: 50%;
  display: inline-block;
  margin-right: 5px;
  box-shadow: 4px 4px 4px grey;
}

.airline-rating-not-filled-in {
  height: 25px;
  width: 25px;
  background-color: lightgray;
  border-radius: 50%;
  display: inline-block;
  margin-right: 5px;
  box-shadow: 4px 4px 4px grey;
}