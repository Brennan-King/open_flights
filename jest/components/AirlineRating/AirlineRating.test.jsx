/* eslint-disable max-len */
import 'jsdom-global/register';
import React from 'react';
import AirlineRating from '../../../app/javascript/components/AirlineRating/AirlineRating';
import {configure, mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({adapter: new Adapter()});

/**
 * Test suite for AirlineRating component
 */
describe('AirlineRating', () => {
  it('should render an AirlineRating component with default props', () => {
    const component = mount(
        <AirlineRating />,
    );

    expect(component).toMatchSnapshot();
  });

  it('should render an AirlineRating component with a three star review', () => {
    const component = mount(
        <AirlineRating
          reviewScore={3}
        />,
    );

    expect(component).toMatchSnapshot();
  });

  it('should render an AirlineRating component with a five star review', () => {
    const component = mount(
        <AirlineRating
          reviewScore={5}
        />,
    );

    expect(component).toMatchSnapshot();
  });
});
