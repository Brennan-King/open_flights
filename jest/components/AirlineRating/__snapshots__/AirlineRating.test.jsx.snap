// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AirlineRating should render an AirlineRating component with a five star review 1`] = `
<AirlineRating
  reviewScore={5}
>
  <span
    className="airline-rating-wrapper"
  >
    <div
      className="airline-rating-filled-in"
    />
    <div
      className="airline-rating-filled-in"
    />
    <div
      className="airline-rating-filled-in"
    />
    <div
      className="airline-rating-filled-in"
    />
    <div
      className="airline-rating-filled-in"
    />
  </span>
</AirlineRating>
`;

exports[`AirlineRating should render an AirlineRating component with a three star review 1`] = `
<AirlineRating
  reviewScore={3}
>
  <span
    className="airline-rating-wrapper"
  >
    <div
      className="airline-rating-filled-in"
    />
    <div
      className="airline-rating-filled-in"
    />
    <div
      className="airline-rating-filled-in"
    />
    <div
      className="airline-rating-not-filled-in"
    />
    <div
      className="airline-rating-not-filled-in"
    />
  </span>
</AirlineRating>
`;

exports[`AirlineRating should render an AirlineRating component with default props 1`] = `
<AirlineRating
  reviewScore={0}
>
  <span
    className="airline-rating-wrapper"
  >
    <div
      className="airline-rating-not-filled-in"
    />
    <div
      className="airline-rating-not-filled-in"
    />
    <div
      className="airline-rating-not-filled-in"
    />
    <div
      className="airline-rating-not-filled-in"
    />
    <div
      className="airline-rating-not-filled-in"
    />
  </span>
</AirlineRating>
`;
